### DISCUSSION TITLE
[Feature Request] persist queryCache in local storage

### DISCUSSION THREAD

**Discussion by huchenme** [TanStack/query#651] [General]
I have been persisting query results in local storage and updating local storage in background in my [chrome extension](https://github.com/huchenme/hacker-tab-extension) manually. As I will start work on supporting multiple languages, it will be nice if `react-query` could provide a native solution/plugin to read/write cache results in local storage.

Related discussion: https://github.com/tannerlinsley/react-query/discussions/649

What is your thought? I am happy to contribute to it if you think it is a nice addition.

**Comment by tannerlinsley** [comment]
I would be *more* open to simply improving the APIs needed to make the process easier for you and others to write your own (at least until for now). Then after that, we can discuss an official package?

**Reply by huchenme** [reply]
that would be great, I am already writing my own solution so I think the API is fine of getting the data. I would love to hear your thoughts on the suggestion of getting the feature natively (maybe one of the options for `ReactQueryConfigProvider` called `useLocalStorage`, if that is true, whatever in `cacheQuery` will automatically persist or read from local storage as well. In that case if user refresh the page, the existing cache will still be there.

**Comment by sudo-vaibhav** [comment]
we have `import { createSyncStoragePersister } from "@tanstack/query-sync-storage-persister";` now
